src/net/sourceforge/kolmafia/session/EquipmentManager.java

```diff
@@ -2131,13 +2131,21 @@ public static final SpecialOutfit currentOutfit() {
   }
 
   public static final SpecialOutfit currentOutfit(AdventureResult[] equipment) {
+    // Go through any outfit that any worn item belongs to.
     int hash = SpecialOutfit.equipmentHash(equipment);
-    for (SpecialOutfit outfit : EquipmentDatabase.normalOutfits.values()) {
-      if (outfit == null) {
-        continue;
-      }
-      if (outfit.isWearing(equipment, hash)) {
-        return outfit;
+    List<Integer> checkedOutfits = new ArrayList<>();
+    for (int i = 0; i < EquipmentManager.FAMILIAR; i++) {
```

Comment:
this sort of thing where we treat the HAT...FAMILIAR stuff as integers always looks dodgy to me (especially as 2 is "holster").

You could start it "for (int i = EquipmentManager.HAT" to make it slightly clearer what's going on. Maybe it's okay as is.

Reply:
I'll make that change, thanks, but yeah, this is a bad pattern.